src/components/basic/StaticTable/VerticalTable.tsx

```diff
@@ -18,51 +18,157 @@
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
 
+import { useState } from 'react'
 import { Typography } from '@mui/material'
-import { type TableType } from './types'
+import EditIcon from '@mui/icons-material/Edit'
+import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline'
+import CloseIcon from '@mui/icons-material/Close'
+import { TableType } from './types'
+import { Input } from '../Input'
+import { Tooltips } from '../ToolTips'
 
-export const VerticalTable = ({ data }: { data: TableType }) => (
-  <table style={{ width: '100%', borderCollapse: 'collapse' }}>
-    <thead>
-      <tr>
-        {data.head.map((col, c) => (
-          <th
-            key={c}
-            style={{
-              backgroundColor: '#ecf0f4',
-              textAlign: 'left',
-              padding: '10px 15px',
+const DOMAIN =
+  /([a-z0-9]|[a-z0-9][a-z0-9-]{0,61}[a-z0-9])(\.([a-z0-9]|[a-z0-9][a-z0-9-]{0,61}[a-z0-9])){1,10}/i
+const URLPATH = /(\/[a-z0-9-._~:/?#[\]@!$&'()*+,;=%]{0,500}){0,20}/
+
+const isURL = (expr: string) =>
+  new RegExp(
+    `^(https)://(${DOMAIN.source})(:\\d{1,5})?(${URLPATH.source})?$`,
+    'i'
+  ).test(expr)
+
+export const VerticalTable = ({
+  data,
+  handleEditURL,
```

Comment:
the entire logic is too specific for a shared component

Reply:
@oyo I have updated the logic and doesnt make depend on shared component anymore. Can you please have a look